base/src/worksheet.rs

```diff
@@ -128,53 +139,101 @@ impl Worksheet {
         Ok(())
     }
 
-    pub fn set_cell_style(&mut self, row: i32, column: i32, style_index: i32) {
+    pub fn set_cell_style(
+        &mut self,
+        row: i32,
+        column: i32,
+        style_index: i32,
+    ) -> Result<(), String> {
         match self.cell_mut(row, column) {
             Some(cell) => {
                 cell.set_style(style_index);
             }
             None => {
-                self.cell_clear_contents_with_style(row, column, style_index);
+                self.cell_clear_contents_with_style(row, column, style_index)?;
             }
         }
+        Ok(())
 
         // TODO: cleanup check if the old cell style is still in use
     }
 
-    pub fn set_cell_with_formula(&mut self, row: i32, column: i32, index: i32, style: i32) {
+    pub fn set_cell_with_formula(
+        &mut self,
+        row: i32,
+        column: i32,
+        index: i32,
+        style: i32,
+    ) -> Result<(), String> {
         let cell = Cell::new_formula(index, style);
-        self.update_cell(row, column, cell);
+        self.update_cell(row, column, cell)?;
+        Ok(())
     }
 
-    pub fn set_cell_with_number(&mut self, row: i32, column: i32, value: f64, style: i32) {
+    pub fn set_cell_with_number(
+        &mut self,
+        row: i32,
+        column: i32,
+        value: f64,
+        style: i32,
+    ) -> Result<(), String> {
         let cell = Cell::new_number(value, style);
-        self.update_cell(row, column, cell);
+        self.update_cell(row, column, cell)?;
+        Ok(())
     }
 
-    pub fn set_cell_with_string(&mut self, row: i32, column: i32, index: i32, style: i32) {
+    pub fn set_cell_with_string(
+        &mut self,
+        row: i32,
+        column: i32,
+        index: i32,
+        style: i32,
+    ) -> Result<(), String> {
         let cell = Cell::new_string(index, style);
-        self.update_cell(row, column, cell);
+        self.update_cell(row, column, cell)?;
+        Ok(())
     }
 
-    pub fn set_cell_with_boolean(&mut self, row: i32, column: i32, value: bool, style: i32) {
+    pub fn set_cell_with_boolean(
+        &mut self,
+        row: i32,
+        column: i32,
+        value: bool,
+        style: i32,
+    ) -> Result<(), String> {
         let cell = Cell::new_boolean(value, style);
-        self.update_cell(row, column, cell);
+        self.update_cell(row, column, cell)?;
+        Ok(())
```

Comment:
`  self.update_cell(row, column, cell)?`

This would not possible as set_cell_with_string function's return type is  Result<(), String>. If add '?' it would result in mismatch of return type.

`self.update_cell(row, column, cell)`

Alternatively, just doing this is correct ( removing Ok(()) ). And I have updated it with that